packages/messaging/src/controllers/sw-controller.ts

```diff
@@ -291,14 +291,37 @@ async function getWindowClient(url: string): Promise<WindowClient | null> {
  * @returns If there is currently a visible WindowClient, this method will
  * resolve to true, otherwise false.
  */
-function hasVisibleClients(clientList: WindowClient[]): boolean {
-  return clientList.some(
-    client =>
-      client.visibilityState === 'visible' &&
-      // Ignore chrome-extension clients as that matches the background pages
-      // of extensions, which are always considered visible for some reason.
-      !client.url.startsWith('chrome-extension://')
+async function hasVisibleClients(clientList: WindowClient[]): Promise<boolean> {
+  const checkedClientList = await Promise.all(
+    clientList.map(
+      async client =>
+        client.visibilityState === 'visible' &&
+        // Ignore browser extension clients as that matches the background pages
+        // of extensions, which are always considered visible for some reason.
+        !(await isBackgroundClient(client))
+    )
   );
+
+  return checkedClientList.some(item => item);
+}
+
+/**
+ * @returns If client is the background page of browser extension, this method will
+ * resolve to true, otherwise false.
+ */
+async function isBackgroundClient(client: WindowClient): Promise<boolean> {
+  const runtime = getBrowserExtensionRuntime();
+
+  if (runtime) {
+    try {
+      const backgroundClient = await runtime.getBackgroundClient();
+      return client === backgroundClient;
+    } catch {
```

Comment:
Enkot,
Can we have a debug log here on the error and error message so it would be easier for folks to debug if they hit this path and wanted to get information?